app/views/programming_groups/index.html.slim

```diff
@@ -0,0 +1,23 @@
+h1 = ProgrammingGroup.model_name.human(count: 2)
+
+= render(layout: 'shared/form_filters') do |f|
+  .col-auto
+    = f.label(:exercise_id, t('activerecord.attributes.programming_group.exercise_title'), class: 'visually-hidden form-label')
+    = f.collection_select(:exercise_id, Exercise.with_programming_groups, :id, :title, class: 'form-control', prompt: t('activerecord.attributes.programming_group.exercise_title'))
+
+.table-responsive
+  table.table.mt-4
+    thead
+      tr
+        th = t('activerecord.attributes.programming_group.programming_group_id')
+        th = t('activerecord.attributes.programming_group.exercise_title')
+        th = t('activerecord.attributes.programming_group.participants')
+        th = t('shared.actions')
+    tbody
+      - @programming_groups.each do |programming_group|
+        tr
+          td = programming_group.displayname
+          td = link_to_if(policy(programming_group.exercise).show?, programming_group.exercise.title, programming_group.exercise, 'data-turbolinks' => "false")
+          td == programming_group.users.map { |user| link_to_if(policy(user).show?, user.name, user) }.join(', ')
```

Comment:
For groups with many users, this might be too much for the column. For these cases, we might think of some good truncation (e.g., via CSS). For now, it might be fine.

Reply:
I mentioned it in issue #1961.